utils.py

```diff
@@ -309,42 +315,45 @@ def canStartBuild(builder, wfb, request):
 
 
 @util.renderer
-def mtrJobsMultiplier(props):
+def mtrJobsMultiplier(props: IProperties) -> int:
     jobs = props.getProperty("jobs", default=20)
     return jobs * 2
 
 
-# ls2string gets the output of ls and returns a space delimited string with the files and directories
-def ls2string(rc, stdout, stderr):
-    lsFilenames = []
-
-    for l in stdout.strip().split("\n"):
-        if l != "":
-            lsFilenames.append(l.strip())
+# ls2string gets the output of ls and returns a space delimited string with
+# the files and directories
+def ls2string(rc: int, stdout: str, stderr: str) -> dict[str, str]:
+    ls_filenames = []
+    for line in stdout.strip().split("\n"):
+        line = line.strip()
+        if not line:
+            ls_filenames.append(line)
 
-    return {"packages": " ".join(lsFilenames)}
+    return {"packages": " ".join(ls_filenames)}
 
 
-# ls2list gets the output of ls and returns a list with the files and directories
-def ls2list(rc, stdout, stderr):
-    lsFilenames = []
+# ls2list gets the output of ls and returns a list with the files and
+# directories
+def ls2list(rc: int, stdout: str, stderr: str) -> dict[str, list[str]]:
+    ls_filenames = []
 
-    for l in stdout.strip().split("\n"):
-        if l != "":
-            lsFilenames.append(l.strip())
+    for line in stdout.strip().split("\n"):
+        line = line.strip()
+        if not line:
+            ls_filenames.append(line)
 
-    return {"packages": lsFilenames}
+    return {"packages": ls_filenames}
 
 
 # Save packages for current branch?
-def savePackage(step, savedBranches=savedPackageBranches):
+def savePackage(step: BuildStep) -> bool:
     return step.getProperty("save_packages") and fnmatch_any(
-        step.getProperty("branch"), savedBranches
+        step.getProperty("branch"), savedPackageBranches
```

Comment:
@cvicentiu Please be aware that `master-galera` has a different value for `savedPackageBranches` and won't be taken into account. 

The `savedPackageBranches` from constants take precedence.

![image](https://github.com/user-attachments/assets/11da85dc-c1e9-47c9-b3b5-ce47603449ce)